src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -447,6 +448,44 @@ internal string AdvanceThroughBalancedParentheticalExpression()
             return expressionText;
         }
 
+        /// <summary>
+        /// Get the current position in this lexer that can be used restore the lexer to this position later.
+        /// </summary>
+        internal object SnapshotPosition()

```

Comment:
>object [](start = 17, length = 6)

The Position should be properly declared if this is something we want to support in the Lexer.